docs/hashicorp_vault.md

```diff
@@ -0,0 +1,115 @@
+---
+id: hashicorp_vault
+title: HashicorpVault
+---
+
+# Hashicorp Vault
+
+The goal of this guide is to configure Signatory to use an Hashicorp Vault as a signing backend.
+
+## **Vault setup**
+
+Run Vault server or use dedicated cloud instance (HCP Vault) for secrets management
+
+### **Install vault**
+
+https://developer.hashicorp.com/vault/downloads
+
+### **Run Dev Server**
+
+```sh
+docker run --cap-add=IPC_LOCK -d --name=dev-vault vault
+```
+
+### **Configure Vault Address and Token**
+
+Set the Vault address and obtain the root token provided when starting the Vault server.
+
+```sh
+export VAULT_ADDR=http://127.0.0.1:8200
+export VAULT_TOKEN={{root_token}}
+```
+
+### **Enable Transit Secrets Engine**
+
+Enable the Transit secrets engine which is used to manage cryptographic functions.
+
+```sh
+vault secrets enable transit
+```
+
+### **Create Transit Key**
+
+Create a new encryption key in the transit secrets engine.
+
+```sh
+vault write -f transit/keys/{{my-key}}
```

Comment:
the default key type will not be discovered by Signatory, this line should read `vault write -f transit/keys/{{my-key}} type="ed25519"`